### PR TITLE
api reviews addressed

### PR DESCRIPTION
**Pull request by moderakh**
Kushagra and I went through some of Jonathan's comments on v4.
This PR captures our work together on addressing some of the comments which could be taken care of quickly.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/AsyncDocumentClient.java [hunk 1]
```diff
@@ -195,7 +196,7 @@ public AsyncDocumentClient build() {
                                                                    connectionPolicy,
                                                                    desiredConsistencyLevel,
                                                                    configs,
-                                                                   tokenResolver,
+                cosmosAuthorizationTokenResolver,
```
Comment (kirankumarkolli) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/AsyncDocumentClient.java]: Indenttaion

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentServiceResponse.java [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 package com.azure.cosmos.implementation;
 
+import com.azure.cosmos.Permission;
```
Comment (kirankumarkolli) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentServiceResponse.java]: undo?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/UniqueKeyPolicy.java [hunk 3]
```diff
@@ -34,7 +34,7 @@ public UniqueKeyPolicy() {
      *
      * @return the unique keys.
      */
-    public Collection<UniqueKey> uniqueKeys() {
+    public List<UniqueKey> setUniqueKeys() {
```
Comment (kirankumarkolli) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/UniqueKeyPolicy.java]: Isn't it supposed to be  getUniqueKeys?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentServiceResponse.java (hunk 2 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentServiceResponse.java]: this is required. as permission is moved from one package to another and hence not accessible in the old package.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/UniqueKeyPolicy.java (hunk 3 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/UniqueKeyPolicy.java]: correct. intelliji refactoring messed up. thanks for catching will fix.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ItemOperations.java [hunk 4]
```diff
@@ -11,7 +11,7 @@
 /**
  * Note: although this class is public it is not part of our public API and may change in future.
  */
-public class ItemOperations {
+final public class ItemOperations {
```
Comment (kirankumarkolli) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ItemOperations.java]: Is it a first class API (in public namespace)? Or will it be internal for now and later we will revisit on how to make it first class?

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ItemOperations.java]: we will move it to implementation package.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/AsyncDocumentClient.java (hunk 1 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/AsyncDocumentClient.java]: will fix. thanks.

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/AsyncDocumentClient.java]: addressed.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/UniqueKeyPolicy.java (hunk 3 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/UniqueKeyPolicy.java]: addressed.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ItemOperations.java (hunk 4 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ItemOperations.java]: moved to implementation.

**Pull request merged**